File: Composicion_y_Agregacion/Ejercicio1/python/Coche.py

from Vehiculo import Vehiculo
class Coche(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, tipo_combustible, num_puertas):  
        super().__init__(marca, modelo, anio, precio_base)   
        self.tipo_combustible = tipo_combustible       
        self.num_puertas = num_puertas  
    
    def mostrar_info(self):
        super().mostrar_info()        
        print(f"Tipo de combustible: {self.tipo_combustible}")
        print(f"Numero de puertas: {self.num_puertas}")
    
    def mas_de_4(self):        
        if self.num_puertas > 4:           
            print(f"- {self.get_marca()} de {self.get_modelo()} tiene {self.num_puertas} puertas.")

File: Herencia/Ejercicio1/Java/Coche.java
package Herencia.Ejercicio1.Java;

public class Coche extends Vehiculo {
    private String tipo_combustible;
    private int num_puertas;

    public Coche(String maa, String mo, int a, int p, String c, int t) {
        super(maa, mo, a, p);
        this.tipo_combustible = c;
        this.num_puertas = t;
    }

    public void mostrar_info() {

    super.mostrar_info();
    System.out.println("Tipo de combustible:" + tipo_combustible); 
    System.out.println("Número de puertas: " + num_puertas);
    }
    
    public void masDe4() {
    if (num_puertas > 4) { 
        System.out.println("-"+ getMarca() + " de " + getModelo() + " tiene " + num_puertas + " puertas.");

        }
    }
}

